src/Appwrite/Migration/Version/V15.php

```diff
@@ -0,0 +1,1294 @@
+                 */
+                $this->projectDB->updateAttribute(
+                    collection: $table,
+                    id: $attribute,
+                    type: Database::VAR_DATETIME,
+                    signed: false
+                );
+            } catch (\Throwable $th) {
+                Console::warning("Add 'datetime' filter to '{$attribute}' from {$table}: {$th->getMessage()}");
+            }
+        }
+
+        $this->projectDB->deleteCachedCollection($table);
+    }
+
+    /**
+     * Create the '_permissions' column to a table.
+     *
+     * @param string $table
+     * @return void
+     * @throws \Exception
+     * @throws \PDOException
+     */
+    protected function createPermissionsColumn(string $table): void
+    {
+        $columns = $this->getSQLColumnTypes($table);
+
+        if (!array_key_exists('_permissions', $columns)) {
+            try {
+                $this->pdo->prepare("ALTER TABLE IF EXISTS `{$this->projectDB->getDefaultDatabase()}`.`_{$this->project->getInternalId()}_{$table}` ADD `_permissions` MEDIUMTEXT DEFAULT NULL")->execute();
+            } catch (\Throwable $th) {
+                Console::warning("Add '_permissions' column to '{$table}': {$th->getMessage()}");
+            }
+        }
+    }
+
+    /**
+     * Populate '$permissions' from '$read' and '$write'.
+     *
+     * @param \Utopia\Database\Document $document
+     * @param null|string $table
+     * @param bool $addCreatePermission
+     * @return void
+     * @throws \Exception
+     * @throws \PDOException
+     */
+    protected function populatePermissionsAttribute(Document &$document, ?string $table = null, bool $addCreatePermission = true): void
+    {
+        $table ??= $document->getCollection();
+
+        $query = $this->pdo->prepare("SELECT * FROM `{$this->projectDB->getDefaultDatabase()}`.`_{$this->project->getInternalId()}_{$table}_perms` WHERE _document = '{$document->getId()}'");
+        $query->execute();
+        $results = $query->fetchAll();
+        $permissions = [];
+
+        foreach ($results as $result) {
+            $type = $result['_type'];
+            $permission = $this->migratePermission($result['_permission']);
+
+            if ($type === 'write') {
+                $permissions[] = "update(\"{$permission}\")";
+                $permissions[] = "delete(\"{$permission}\")";
+                if ($addCreatePermission) {
+                    $permissions[] = "create(\"{$permission}\")";
+                }
+            } else {
+                $permissions[] = "{$type}(\"{$permission}\")";
+            }
+        }
+
+        $document->setAttribute('$permissions', $permissions);
+    }
+
+    /**
+     * Migrates a permission string
+     *
+     * @param string $permission
+     * @return string
+     */
+    protected function migratePermission(string $permission): string
+    {
+        return match ($permission) {
+            'role:all' => 'any',
```

Comment:
Do we want to do this as it would change the behavior to write operations between 0.15 and 1.0 (`role:all` for write operations in 0.15 is actually `role:member`).

Reply:
iirc role:all also includes guests right?

talking about role:member, did it stay the same?

Reply:
`role:all` includes guests, but it doesn't actually give guests access for write operations. For example, in 0.15, if you put `role:all` for function execute permission, a guest still can't actually execute the function.

Reply:
ah okay! gotcha 👍🏻 

Gonna add this case for write permission.